src/Spd.Resource.Repository/BizContact/BizContactRepository.cs

```diff
@@ -0,0 +1,100 @@
+        private readonly DynamicsContext _context;
+        private readonly IMapper _mapper;
+        private readonly ILogger<BizContactRepository> _logger;
+
+        public BizContactRepository(IDynamicsContextFactory ctx,
+            IMapper mapper,
+            ILogger<BizContactRepository> logger)
+        {
+            _context = ctx.CreateChangeOverwrite();
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<BizContactResp>> GetBizAppContactsAsync(BizContactQry qry, CancellationToken ct)
+        {
+            IQueryable<spd_businesscontact> bizContacts = _context.spd_businesscontacts;
+            if (!qry.IncludeInactive)
+                bizContacts = bizContacts.Where(a => a.statecode != DynamicsConstants.StateCode_Inactive);
+            if (qry.BizId != null)
+                bizContacts = bizContacts.Where(a => a._spd_organizationid_value == qry.BizId);
+            if (qry.AppId != null)
+                bizContacts = bizContacts.Where(a => a._spd_application_value == qry.AppId);
+            if (qry.RoleCode != null)
+                bizContacts = bizContacts.Where(a => a.spd_role == (int?)SharedMappingFuncs.GetOptionset<BizContactRoleEnum, BizContactRoleOptionSet>(qry.RoleCode));
+
+            return _mapper.Map<IEnumerable<BizContactResp>>(bizContacts.ToList());
+        }
+
+        public async Task<Unit> ManageBizContactsAsync(BizContactUpsertCmd cmd, CancellationToken ct)
+        {
+            IQueryable<spd_businesscontact> bizContacts = _context.spd_businesscontacts;
+            if (cmd.BizId != null)
+                bizContacts = bizContacts.Where(a => a._spd_organizationid_value == cmd.BizId);
+            if (cmd.AppId != null)
+                bizContacts = bizContacts.Where(a => a._spd_application_value == cmd.AppId);
+            var list = bizContacts.ToList();
+            //remove all not in cmd.Data
+            var toRemove = list.Where(c => !cmd.Data.Any(d => d.BizContactId == c.spd_businesscontactid));
+            foreach (var item in toRemove)
+            {
+                item.statecode = DynamicsConstants.StateCode_Inactive;
+                item.statuscode = DynamicsConstants.StatusCode_Inactive;
+                _context.UpdateObject(item);
+            }
+
+            //update all that in cmd.Data
+            var toModify = list.Where(c => cmd.Data.Any(d => d.BizContactId == c.spd_businesscontactid));
+            foreach (var item in toModify)
+            {
+                _mapper.Map(cmd.Data.FirstOrDefault(d => d.BizContactId == item.spd_businesscontactid), item);
+                _context.UpdateObject(item);
+            }
+
+            //add all that in cmd.Data which does not have id
+            var toAdd = cmd.Data.Where(d => d.BizContactId == null).ToList();
+            if (toAdd.Count > 0)
+            {
+                account? biz = _context.accounts.Where(a => a.accountid == cmd.BizId).FirstOrDefault();

```

Comment:
We will have to be careful when we do this type of query, because it could throw "not found" exception. We can avoid that by adding `statecode == DynamicsConstants.StateCode_Active` as an additional condition